spec/compiler.js

```diff
@@ -108,4 +108,29 @@ describe('compiler', function() {
       equal(/return ""/.test(Handlebars.precompile('')), true);
     });
   });
+
+  describe('Compile Partials Correctly', function() {
+    var compileCount = 0;
+    var origTemplateFunc = Handlebars.template;
+    before(function() {
+      Handlebars.template = function(params) {
+        compileCount++;
+        return origTemplateFunc(params);
+      };
+      Handlebars.registerPartial({
+        'dude': 'I am a partial'
+      });
+    });
+    after(function() {
+      Handlebars.template = origTemplateFunc;
+      Handlebars.unregisterPartial('dude');
```

Comment:
If think it is better to use "Handlebars.create()" to create a new instance (in the `it`-block) instead.
Please try if this works.